include/zephyr/init.h

```diff
@@ -21,13 +21,14 @@ extern "C" {
  * stack. The remaining levels are executed in the kernel's main task.
  */
 
-#define _SYS_INIT_LEVEL_PRE_KERNEL_1	0
-#define _SYS_INIT_LEVEL_PRE_KERNEL_2	1
-#define _SYS_INIT_LEVEL_POST_KERNEL	2
-#define _SYS_INIT_LEVEL_APPLICATION	3
+#define _SYS_INIT_LEVEL_ARCH		0
+#define _SYS_INIT_LEVEL_PRE_KERNEL_1	1
+#define _SYS_INIT_LEVEL_PRE_KERNEL_2	2
+#define _SYS_INIT_LEVEL_POST_KERNEL	3
+#define _SYS_INIT_LEVEL_APPLICATION	4
 
 #ifdef CONFIG_SMP
-#define _SYS_INIT_LEVEL_SMP		4
+#define _SYS_INIT_LEVEL_SMP		5
```

Comment:
Define these in terms of the level before.  For example:

`#define _SYS_INIT_LEVEL_SMP (_SYS_INIT_LEVEL_APPLICATION + 1)`